src/db/statement.cr

```diff
@@ -0,0 +1,112 @@
+module DB
+  # Represents a prepared query in a `Connection`.
+  # It should be created by `QueryMethods`.
+  #
+  # ### Note to implementors
+  #
+  # 1. Subclass `Statements`
+  # 2. `Statements` are created from a custom driver `Connection#prepare` method.
+  # 3. `#perform_query` executes a query that is expected to return a `ResultSet`
+  # 4. `#perform_exec` executes a query that is expected to return an `ExecResult`
+  # 6. `#do_close` is called to release the statement resources.
+  abstract class Statement
+    include Disposable
+
+    # :nodoc:
+    getter connection
+
+    def initialize(@connection)
+    end
+
+    protected def do_close
```

Comment:
Maybe make this abstract so we force its implementation, even if it ends up empty?